README.md

```diff
@@ -1,2 +1,233 @@
+   - [rm](#rm)
+   - [cp](#cp)
+   - [mv](#mv)
+   - [get](#get)
+   - [get_last_object](#get_last_object)
+   - [put_data](#put_data)
+   - [put_file](#put_file)
+4. [Contribute](#Contribute)
+
+## Installation
+Use the package manager [pip](https://pypi.org/project/pyminio/) to install pyminio.
+```bash
+pip install pyminio
+```
+
+## Setting up Pyminio:
+In case you want to add your own minio object you can pass it in the constructor like so:
+
+```python
+from minio import Minio
+from pyminio import Pyminio
+
+ENDPOINT = environ.get('MINIO_CONNECTION')
+ACCESS_KEY = environ.get('MINIO_ACCESS_KEY')
+SECRET_KEY = environ.get('MINIO_SECRET_KEY')
+
+minio_obj = Minio(
+            endpoint=self.ENDPOINT,
+            access_key=self.ACCESS_KEY,
+            secret_key=self.SECRET_KEY
+        )
+pyminio_client = Pyminio(minio_obj=minio_obj)
+```
+
+if you dont want to handle with minio, you cand do this instead:
+
+```python
+from pyminio import Pyminio
+
+ENDPOINT = environ.get('MINIO_CONNECTION')
+ACCESS_KEY = environ.get('MINIO_ACCESS_KEY')
+SECRET_KEY = environ.get('MINIO_SECRET_KEY')
+
+pyminio_client = Pyminio.from_credentials(
+    endpoint=self.ENDPOINT,
+    access_key=self.ACCESS_KEY,
+    secret_key=self.SECRET_KEY
+)
+```
+
+You can storage your minio credentials in environment varibles like in these examples or pass them to Pyminio in any other way.
+
+## Usage:
+
+### mkdirs
+Pyminio.mkdirs will create the given full path if not exists like linux's 'mkdir -p'.
+
+This function must get a directory path or it will raise a ValueError.
+
+```python
+>>> pyminio_client.mkdirs('/foo/bar/baz/')
+```
+
+### listdir
+Pyminio.listdir will return the directory's content as a list of directorys and files name. Works like os's 'listdir'.
+
+This function must get a directory path or it will raise a ValueError.
+
+```python
+>>> pyminio_client.listdir('/foo/bar/baz/')
+['file_name_1', 'file_name_2', 'directory_name/']
+```
+
+There is an option to use the only_files flag to get only files from listdir.
+
+```python
+>>> pyminio_client.listdir('/foo/bar/baz/', only_files=True)
+['file_name_1', 'file_name_2']
+```
+
+### exists
+Pyminio.exists will return a boolean that confirm reather this path exists or not in the server. Works like os.path.exists.
+
+```python
+>>> pyminio_client.exists('/foo/bar/baz/file_name_1')  # existed file
+True
+>>> pyminio_client.exists('/foo/bar/baz/file_name_3')  # not existed file
+False
+>>> pyminio_client.exists('/all/path/wrong/')  # not existed path
+False
+```
+
+### isdir
+Pyminio.isdir will return True only if the given path exists and is a directory. Works like os.path.isdir
+
+```python
+>>> pyminio_client.isdir('/foo/bar/baz/file_name_1')  # existed file
+False
+>>> pyminio_client.isdir('/foo/bar/baz/directory_name/')  # existed directory
+True
+>>> pyminio_client.isdir('/all/path/wrong/but/directory/')  # not existed directory
+False
+```
+
+### truncate
+Pyminio.truncate will delete all minio's content.
```

Comment:
I don't understand what it does from this explanation
also, truncate is a pretty confusing verb, so think if it fits

Reply:
truncate actually is very known from SQL worlds.
maybe void could be a better name :\ I dont think so.
got a suggestion?